### DISCUSSION TITLE
Custom Meeting type

### DISCUSSION THREAD

**Discussion by eflashlogistics** [code4recovery/12-step-meeting-list#1419] [Q&A] [answered]
Greetings

How do we add multiple custom meeting types? I added the snippit to functions.php and it works. But was lost on adding a second. Thanks

**Comment by joshreisner** [comment, marked as answer]
you can add more lines inside the array. be sure to include the commas, or you could get a PHP error

```
if (function_exists('tsml_custom_types')) {
    tsml_custom_types(array(
        'XYZ' => 'My Custom Type',
        'YZA' => 'My Second Custom Type',
        'ZAB' => 'My Third Custom Type',
    ));
}
```

**Reply by eflashlogistics** [reply]
Ok thats what I did wrong. I get it. 

May I ask about this statement? "Be careful with the codes (“XYZ” in the above example) as this gives you the ability to replace existing types."

For the other codes such as YZA, ZAB, can we just choose these randomly? Will some codes overwrite existing types? If so how I know what codes not to use?

Thanks

**Reply by joshreisner** [reply]
sure, here's a list of all the AA meeting types: https://github.com/code4recovery/spec?tab=readme-ov-file#meeting-types

i'd also try to stay away from any unimplemented language codes, eg `ZH` which you can do by making your code three or more letters

we don't add types often, so in all likelihood it probably won't be a problem

**Comment by kiyote33** [comment]
Hi @eflashlogistics,
So you might extend your snippet something like this: 
`if (function_exists('tsml_custom_types')) {
    tsml_custom_types(array(
        'XYZ' => 'My Custom Type',
        'ABC' => 'My Second snippet.
    ));
}`
